pkg/server/v3/server_test.go

```diff
@@ -61,7 +63,52 @@ func (config *mockConfigWatcher) CreateWatch(req *discovery.DiscoveryRequest) (c
 }
 
 func (config *mockConfigWatcher) CreateDeltaWatch(req *discovery.DeltaDiscoveryRequest, st *stream.StreamState) (chan cache.DeltaResponse, func()) {
-	return nil, nil
+	config.counts[req.TypeUrl] = config.counts[req.TypeUrl] + 1
+
+	// Create our out watch channel to return with a buffer of one
+	out := make(chan cache.DeltaResponse, 1)
+
+	if len(config.deltaResponses[req.TypeUrl]) > 0 {
+		res := config.deltaResponses[req.TypeUrl][0]
+		var subscribed []types.Resource
+
+		// Only return back the subscribed resources to our request type
+		r, _ := res.GetDeltaDiscoveryResponse()
+		if len(req.GetResourceNamesSubscribe()) != 0 {
+			for _, resource := range r.Resources {
+				for _, alias := range req.GetResourceNamesSubscribe() {
+					if resource.GetName() == alias {
+						subscribed = append(subscribed, resource)
+					}
+				}
+			}
+		} else {
+			// We do this to handle the wildcard situation (just return all for testing)
+			for _, resource := range r.Resources {
+				subscribed = append(subscribed, resource)
+			}
+		}
+
+		// We should only send back subscribed resources here
+		out <- &cache.RawDeltaResponse{
+			DeltaRequest:      req,
+			Resources:         subscribed,
+			SystemVersionInfo: "",
+			NextVersionMap:    st.ResourceVersions,
```

Comment:
Don't we need to update this?

Reply:
Actually yes we do, I'm not sure why we don't there. I need to recalculate the hashes, I'll change that. Thank you!